src/__tests__/index.ts

```diff
@@ -358,3 +358,36 @@ describe("Scope branch", () => {
     expect(currenBranch).toBe("release/v1.20200510.3");
   });
 });
+
+describe("when patching with iterateOnScope set to true", () => {
+  it("The first commit to patch a scope needs to find the newest sub scope", () => {
+    // Setup
+    const cwd = createNewRepo();
+
+    createCommit(cwd);
+    tag(`scope/v1.20231211.1.0`, cwd);
+    const scopeTag = "scope/package";
+
+    //Act
+    setVersion({ cwd, scopeTag, patch: true, iterateOnScope: true });
+
+    const tags = getTags(cwd);
+    expect(tags[0]).toContain("scope/package/v1.20231211.1.1");
+  });
+
+  it("All the non-first commits to patch a scope needs should use iterate on their respective scope", () => {
+    // Setup
+    const cwd = createNewRepo();
+
+    createCommit(cwd);
+    tag(`scope/v1.20231211.1.0`, cwd);
+    tag(`scope/package/v1.20231211.1.1`, cwd);
```

Comment:
In reality you would not have these two tags pointing to the same commit. We should setup the two tags on two different commits to get the tests closer to how this library is consumed.